### PR TITLE
Add svg sprites

### PR DESCRIPTION
**Pull request by huwshimi**
Use svg sprites instead of pngs.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- run `make clean-all fast-babel gui run`
- check that everything is built without errors
- view the gui
- check that there are icons in the places they used to be

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3440/
Test PASSed.

File: jujugui/static/gui/src/app/components/svg-icon/svg-icon.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    /**
+      Generates the viewbox for the icon based on the supplied props.
+
+      @method _generateViewbox
+      @returns {String} The viewbox.
+    */
+    _generateViewbox: function() {
+      var size = this.props.size;
+      var width = this.props.width ? this.props.width : size;
+      var height = this.props.height ? this.props.height : size;
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/svg-icon/svg-icon.js]: you do this dance in two different functions, what about storing it in state from `getInitialState`?

File: jujugui/templates/index.html.mako [hunk 2]
```diff
@@ -85,13 +85,21 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
   </head>
 
   <body>
+      <div class="hidden">
+        ## Include the svg sprite image inside the body.
+        <%include file="../static/gui/build/app/assets/stack/svg/sprite.css.svg" />
```
Comment (hatched) [inline jujugui/templates/index.html.mako]: Does this create a link? Or is it embedded?

File: package.json [hunk 3]
```diff
@@ -35,13 +34,13 @@
     "node-markdown": "0.1.x",
     "node-minify": "0.7.x",
```
Comment (hatched) [inline package.json]: I thought that this was used by spritesheet?

File: package.json [hunk 4]
```diff
@@ -35,13 +34,13 @@
...
     "node-sass": "^3.1.2",
-    "node-spritesheet": "0.4.0",
     "phantomjs": "^1.9.7-15",
     "react": "0.14.0",
     "react-dom": "0.14.0",
     "react-onclickoutside": "4.0.1",
     "should": ">=1.0.0",
     "smash": "0.0.12",
```
Comment (hatched) [inline package.json]: I thought that this was used by spritesheet?

File: jujugui/templates/index.html.mako (hunk 2 above)
Comment (huwshimi) [inline jujugui/templates/index.html.mako]: This embeds the file into the html, this is so we can reference it with `<use xlink:href=...`

**Comment by hatched** [comment]
:+1: QA :ok: 
This is great, now that we no longer rely on that library I can go ahead and update node/npm and all the great things that come with that :)

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3442/
Test PASSed.

**Comment by makyo** [comment]
:+1:

**Comment by huwshimi** [comment]
Thanks for the reviews! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/1180

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/1182

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**